tests/badsymbols.pl

```diff
@@ -90,6 +90,9 @@ sub scanenums {
         if($skipit) {
             next;
         }
+        if (/^\#\s*\z/) {
```

Comment:
```suggestion
        if (/^#(line|) (\d+)( |$)/) {
```

Here is the first erroneous entry from the reporter's verbose output ([badsymbols.log](https://github.com/curl/curl/files/5726899/badsymbols.log)):

~~~
Source: /opt/developerstudio12.6/bin/cc -E -I/opt/bw/include -D_POSIX_PTHREAD_SEMANTICS -D_LARGEFILE64_SOURCE -D_TS_ERRNO -I/opt/bw/include ../include/curl/curl.h
Symbol: #
Line #5865: # 488
~~~

All the bad entries are basically the same, the preprocessor shows the line number of the symbol (eg 488) but not the header filename. We expect at least:
~~~
# <line-number-in-header-filename> <header-filename>
  the_line
~~~

https://github.com/curl/curl/blob/e052859759b34d0e05ce0f17244873e5cd7b457b/tests/badsymbols.pl#L70-L81

Since there isn't a filename I would assume that whatever the last filename found is what is used, and based on my review of the log that seems to be the case, so I suspect this is what it's doing:

cc
~~~
# 123 foo.h
  code
# 456
  more_code_from_same_file
~~~

gcc (what we expect, for comparison)
~~~
# 123 foo.h
  code
# 456 foo.h
  more_code_from_same_file
~~~

Reply:
How about this simplified version? Should deal with:

    #
    # 123
    #line 123

~~~perl
if (/^\#(line|) *\d+\z/) {
~~~

Reply:
> ```perl
> /^#(line|) *\d*\z/
> ```

`^#(line|) *\d*$` tests ok at https://regex101.com/ using pcre. The \z isn't going to work right unless you chomp the line I think. IMO $ is easier to understand

Reply:
Hm, so maybe going much simpler, since the only `#` pattern we actually support is already checked, we can skip all lines starting with hash:

        if (/^#/) {
            next;
        }

Reply:
HMMMM. That seems too easy but yes I think you're right.